tools/dev/experiment/values-micro-services-experiment.yaml

```diff
@@ -0,0 +1,71 @@
+pyroscope:
+  extraEnvVars:
+    PYROSCOPE_V2_EXPERIMENT: 1
+  extraArgs:
+    query-backend.address: "dns:///_grpc._tcp.pyroscope-query-worker-headless.$(NAMESPACE_FQDN):9095"
+    metastore.address: "dns:///_grpc._tcp.pyroscope-metastore-headless.$(NAMESPACE_FQDN):9095"
+    metastore.raft.bind-address: ":9099"
+    metastore.raft.server-id: "$(POD_NAME).pyroscope-metastore-headless.$(NAMESPACE_FQDN)"
+    metastore.raft.advertise-address: "$(POD_NAME).pyroscope-metastore-headless.$(NAMESPACE_FQDN):9099"
+    metastore.raft.bootstrap-peers: "dnssrvnoa+_raft._tcp.pyroscope-metastore-headless.$(NAMESPACE_FQDN):9099"
+    metastore.raft.bootstrap-expect-peers: "3"
+
+  components:
+    distributor:
+      kind: Deployment
+      replicaCount: 2
+
+    segment-writer:
+      kind: StatefulSet
+      replicaCount: 3
+      persistence:
+        enabled: false
+
+    query-frontend:
+      kind: Deployment
+      replicaCount: 2
+
+    query-backend:
+      kind: Deployment
+      replicaCount: 3
+      service:
+        extraPorts:
+          - name: grpc
+            port: 9095
+            protocol: TCP
+            targetPort: 9095
+
+    metastore:
+      kind: StatefulSet
+      replicaCount: 3
+      service:
+        publishNotReadyAddresses: true
+        extraPorts:
+          - name: grpc
+            port: 9095
+            protocol: TCP
+            targetPort: 9095
+          - name: raft
+            port: 9099
+            protocol: TCP
+            targetPort: 9099
+      persistence:
+        enabled: true
+        size: 10Gi
+      extraVolumeMounts:
+        - name: data
+          mountPath: /data-metastore
+          subPath: metastore
+
+    compaction-worker:
+      kind: StatefulSet
+      replicaCount: 3
+      persistence:
+        enabled: false
+      extraVolumeMounts:
+        - name: data
+          mountPath: /data-compaction-worker
+          subPath: compaction-worker
```

Comment:
Note that `compaction-worker` is a statefulset: it will be using own persistent disks to avoid impacting the host